Experiences/Experiences/NewExperienceViewController.swift

```diff
@@ -0,0 +1,205 @@
+        case true:
+            saveButton.isEnabled = false
+        case false:
+            saveButton.isEnabled = true
+        }
+    }
+}
+
+extension NewExperienceViewController: AVAudioRecorderDelegate {
+    func audioRecorderDidFinishRecording(_ recorder: AVAudioRecorder, successfully flag: Bool) {
+        if let recordingURL = audioRecorderController.recordingURL {
+            print("Finished recording: \(recordingURL.path)")
+            
+            audioURL = recordingURL
+        } else {
+            print("Did not successfully finish recording")
+        }
+        
+        DispatchQueue.main.async {
+            self.updateViews()
+        }
+    }
+    
+    func audioRecorderEncodeErrorDidOccur(_ recorder: AVAudioRecorder, error: Error?) {
+        if let error = error {
+            NSLog("Error occured during recording: \(error)")
+        }
+        
+        DispatchQueue.main.async {
+            self.updateViews()
+        }
+    }
+}
+
+extension NewExperienceViewController: AudioRecorderControllerDelegate {
+    func didNotRecievePermission() {
+        print("Microphone access has been blocked.")
+        
+        let alertController = UIAlertController(title: "Microphone Access Denied", message: "Please allow this app to access your Microphone.", preferredStyle: .alert)
+        
+        alertController.addAction(UIAlertAction(title: "Open Settings", style: .default) { (_) in
+            UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!)
+        })
+        
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .default, handler: nil))
+        
+        present(alertController, animated: true)
+    }
+}
+
+extension NewExperienceViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        if let image = info[.originalImage] as? UIImage {
+            
+            imageView.image = filterImage(image)
+        }
+        dismiss(animated: true)
+    }
+    
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        dismiss(animated: true)
+    }
+}
+
+extension NewExperienceViewController: CLLocationManagerDelegate {
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+         print("Error getting location: \(error)")
+    }
+
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+        if status == .authorizedWhenInUse {
+            locationManager?.requestLocation()
+        } else if status == .authorizedAlways {
+            locationManager?.requestLocation()
+        }
+    }
+
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        
+        saveExperience()
+    }
+}
```

Comment:
I would try to add a function for the keyboard to hide when finishing editing the textField for the picture, because once you're done creating a name for the picture you cannot see the recoding anymore.